clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2230,7 +2230,7 @@ public void testInvalidGroupMetadata() throws InterruptedException {
         MockClient client = new MockClient(time, metadata);
         initMetadata(client, Collections.singletonMap(topic, 1));
         KafkaConsumer<String, String> consumer = newConsumer(time, client, subscription, metadata,
-                new RoundRobinAssignor(), true, groupInstanceId);
+                new RoundRobinAssignor(), false, groupInstanceId);
```

Comment:
I understand the mockTime change below. But why should we change the `autoCommit` flag here?

Reply:
Because this testInvalidGroupMetadata deliberately calls `new MockTime()` to make `autoTickMs=0`, I cannot set autoTickMs to non-zero here.
`() -> consumer.poll(Duration.ofSeconds(5))` this line of code will block, so that `assertThrows(ConcurrentModificationException.class,consumer::groupMetadata)` can pass,
Then call `consumer.wakeup()` to make `() -> consumer.poll(Duration.ofSeconds(5))` throw `WakeupException` and exit blocking.
`assertTrue(service.awaitTermination(10, TimeUnit.SECONDS))` can also be passed

Reply:
Since the `ConsumerCoordinator#close` method will call `maybeAutoCommitOffsetsAsync()`, we need to add an additional OffsetCommit prepareResponse in `testCloseManualAssignment` case.
The code has been submitted, please help check it, thanks @showuon

Reply:
I'm afraid the change of `autoCommit` flag will change the test purpose. I checked and it looks like we'll wait for consumer to close forever, because the timer didn't tick. Could we do `consumer.close(Duration.ZERO);`, and keep the `autoCommit` as true as before? What do you think?